witnesscalc_adapter/src/lib.rs

```diff
@@ -144,79 +149,77 @@ pub fn build_and_link(circuits_dir: &str) {
+        //For each .cpp file, do the following: find the last include statement (should be #include "calcwit.hpp") and insert the following on the next line: namespace CIRCUIT_NAME {. Then, insert the closing } at the end of the file:
+        let circuit_cpp = fs::read_to_string(&path).expect("Failed to read circuit .cpp file");
+        let circuit_cpp = circuit_cpp.replace(
+            "#include \"calcwit.hpp\"",
+            "#include \"calcwit.hpp\"\nnamespace CIRCUIT_NAME {",
+        );
+        let circuit_cpp = circuit_cpp + "\n}";
+        let circuit_cpp_name = witnesscalc_path.join("src").join(circuit_name);
+        let circuit_cpp_dest = circuit_cpp_name.with_extension("cpp");
+        fs::write(&circuit_cpp_dest, circuit_cpp).expect("Failed to write circuit .cpp file");
+
+        //Find a witnesscalc_template.cpp template file in the src. Replace all the @CIRCUIT_NAME@ inside it with the circuit name and write it to the src directory, replacing "template" in the name with the circuit name
+        let template_path = witnesscalc_path
+            .join("src")
+            .join("witnesscalc_template.cpp");
+        let template = fs::read_to_string(&template_path).expect("Failed to read template file");
+        let template = template.replace("@CIRCUIT_NAME@", circuit_name);
+        let template_dest = witnesscalc_path
+            .join("src")
+            .join(format!("witnesscalc_{}.cpp", circuit_name));
+        fs::write(&template_dest, template).expect("Failed to write the templated .cpp file");
+        //Find a witnesscalc_template.h template file in the src. Replace all the @CIRCUIT_NAME@ inside it with the circuit name, @CIRCUIT_NAME_CAPS@ with the capitalized name, and write it to the src directory, replacing "template" in the name with the circuit name
+        let template_path = witnesscalc_path.join("src").join("witnesscalc_template.h");
+        let template = fs::read_to_string(&template_path).expect("Failed to read template file");
+        let template = template
+            .replace("@CIRCUIT_NAME@", circuit_name)
+            .replace("@CIRCUIT_NAME_CAPS@", &circuit_name.to_uppercase());
+        let template_dest = witnesscalc_path
+            .join("src")
+            .join(format!("witnesscalc_{}.h", circuit_name));
+        fs::write(&template_dest, template).expect("Failed to write the templated .h file");
+    });
+
+    //the circuit name list would look like "circuit1;circuit2;circuit3"
+    let circuit_names = circuit_files
+        .iter()
+        .map(|path| path.file_stem().unwrap().to_str().unwrap())
+        .collect::<Vec<_>>();
+
+    let circuit_names_semicolon = circuit_names.join(";");
+
+    Command::new("make")
+        .env("CIRCUIT_NAMES", circuit_names_semicolon)
+        .arg(witnesscalc_build_target)
+        .current_dir(&witnesscalc_path)
+        .spawn()
+        .expect("Failed to spawn make arm64_host")
```

Comment:
There are a few old error messages here with hardcoded "arm64_host" that we can rewrite.